### PR TITLE
Fork/upgrade to 1.0.10

### PR DESCRIPTION
**Pull request by wmmihaa**
Copied 1.0.10 from thud => zeus

### CODE REVIEW

File: recipes-core/iotedge-daemon/iotedge-daemon.inc
Comment (mschwan-phytec) [inline recipes-core/iotedge-daemon/iotedge-daemon.inc:44]: You may remove this unnecessary whitespace.

**Review by mschwan-phytec — COMMENTED** [review]
In general, this looks good to me, though I did not build nor tested this version.

Looking at previous upgrades, you can also remove any older versions of iotedge with this PR.

**Comment by wmmihaa** [comment]
Removed 1.0.9.4 versions and some whitespace

**Comment by wmmihaa** [comment]
As for testing... I've ran the iotedge check and deployed some custom modules and all seems to work:

**Comment by wmmihaa** [comment]
```
12345678901:/home/mikael# iotedge check
Configuration checks
--------------------
√ config.yaml is well-formed - OK
√ config.yaml has well-formed connection string - OK
√ container engine is installed and functional - OK
‼ config.yaml has correct hostname - Warning
    config.yaml has hostname 12345678901 which does not comply with RFC 1035.

    - Hostname must be between 1 and 255 octets inclusive.
    - Each label in the hostname (component separated by ".") must be between 1 and 63 octets inclusive.
    - Each label must start with an ASCII alphabet character (a-z, A-Z), end with an ASCII alphanumeric character (a-z, A-Z, 0-9), and must contain only ASCII alphanumeric characters or hyphens (a-z, A-Z, 0-9, "-").

    Not complying with RFC 1035 may cause errors during the TLS handshake with modules and downstream devices.
√ config.yaml has correct URIs for daemon mgmt endpoint - OK
‼ latest security daemon - Warning
    Installed IoT Edge daemon has version 1.0.10 but 1.0.10.2 is the latest stable version available.
    Please see https://aka.ms/iotedge-update-runtime for update instructions.
√ host time is close to real time - OK
√ container time is close to host time - OK
‼ DNS server - Warning
    Container engine is not configured with DNS server setting, which may impact connectivity to IoT Hub.
    Please see https://aka.ms/iotedge-prod-checklist-dns for best practices.
    You can ignore this warning if you are setting DNS server per module in the Edge deployment.
‼ production readiness: certificates - Warning
    The Edge device is using self-signed automatically-generated development certificates.
    They will expire in 89 days (at 2021-03-09 22:16:25 UTC) causing module-to-module and downstream device communication to fail on an active deployment.
    After the certs have expired, restarting the IoT Edge daemon will trigger it to generate new development certs.
    Please consider using production certificates instead. See https://aka.ms/iotedge-prod-checklist-certs for best practices.
‼ production readiness: container engine - Warning
    Device is not using a production-supported container engine (moby-engine).
    Please see https://aka.ms/iotedge-prod-checklist-moby for details.
‼ production readiness: logs policy - Warning
    Container engine is not configured to rotate module logs which may cause it run out of disk space.
    Please see https://aka.ms/iotedge-prod-checklist-logs for best practices.
    You can ignore this warning if you are setting log policy per module in the Edge deployment.
‼ production readiness: Edge Agent's storage directory is persisted on the host filesystem - Warning
    The edgeAgent module is not configured to persist its /tmp/edgeAgent directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.
‼ production readiness: Edge Hub's storage directory is persisted on the host filesystem - Warning
    The edgeHub module is not configured to persist its /tmp/edgeHub directory on the host filesystem.
    Data might be lost if the module is deleted or updated.
    Please see https://aka.ms/iotedge-storage-host for best practices.

Connectivity checks
-------------------
√ host can connect to and perform TLS handshake with IoT Hub AMQP port - OK
√ host can connect to and perform TLS handshake with IoT Hub HTTPS / WebSockets port - OK
√ host can connect to and perform TLS handshake with IoT Hub MQTT port - OK
√ container on the default network can connect to IoT Hub AMQP port - OK
√ container on the default network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the default network can connect to IoT Hub MQTT port - OK
√ container on the IoT Edge module network can connect to IoT Hub AMQP port - OK
√ container on the IoT Edge module network can connect to IoT Hub HTTPS / WebSockets port - OK
√ container on the IoT Edge module network can connect to IoT Hub MQTT port - OK

15 check(s) succeeded.
8 check(s) raised warnings. Re-run with --verbose for more details.
```

File: recipes-core/iotedge-daemon/iotedge-daemon_1.0.10.bb
Comment (mschwan-phytec) [inline recipes-core/iotedge-daemon/iotedge-daemon_1.0.10.bb:263]: This is a style change and should reside in a separated commit. Same for iotedge-cli. In the end you should have two commits, one that upgrades all iotedge recipes to 1.0.10, and another one that may fix styling issues.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
Thanks for verifying the upgrade!

You should squash your changes into a single commit. This way meta-iotedge can be checked out at any commit and be built without breakage. Style changes should still be separated, because they do not belong to the upgrade logically.

The file `recipes-core/iotedge-daemon/iotedge-daemon.inc` does not require to be changed and should be left untouched. After squashing you can use `git reset HEAD~1 recipes-core/iotedge-daemon/iotedge-daemon.inc` and `git commit --amend` (I think, please verify before execution) to do that.

File: recipes-core/iotedge-daemon/iotedge-daemon_1.0.10.bb
Comment (wmmihaa) [reply recipes-core/iotedge-daemon/iotedge-daemon_1.0.10.bb]: You previously asked me to remove the whitespaces in `recipes-core/iotedge-daemon/iotedge-daemon.inc`...
> You may remove this unnecessary whitespace.

Also...I had some difficulties squashing the changes, mainly due to the fact that I forked the zeus branch before my previous pr was approved. Would it be ok to disregard this pr entirely and I'll create a new one? If you want me to do that, would you like `recipes-core/iotedge-daemon/iotedge-daemon.inc` w/ or w/o line-breaks?

**Comment by mschwan-phytec** [comment]
By removing the whitespace I meant the two spaces that were added in the first version of your PR. Sorry, if this got a bit confusing. Essentially, it is best to not change the file formatting, except you want to fix some styling issues.

I do not see a reason to discard this PR and open a new one. What you should do is rebase your branch onto the current zeus upstream branch (use `git rebase upstream/zeus` for that). You may have to fix some conflicts then, but nothing that couldn't be done. Squashing the commits seems inevitable by now, as your patch stack grew to 8 commits and repeatedly make the same changes.

Btw, these are just my two cents. I am not maintaining this repository. :wink: In the end, the maintainer of meta-iotedge should have the last word here.

**Comment by micahl** [comment]
@wmmihaa is this PR something you still want to pursue?

**Pull request closed without merging**
